ml-proto/spec/check.ml

```diff
@@ -90,10 +90,17 @@ let type_cvt at = function
     | DemoteFloat64 -> error at "invalid conversion"
     ), Float64Type
 
+type type_hostop_type = {
```

Comment:
Instead of introducing another variant of function type for a singular use, can't `type_hostop` just return a pair `func_type * bool`?